sklearn/metrics/tests/test_score_objects.py

```diff
@@ -459,18 +460,43 @@ def test_supervised_cluster_scorers():
         assert_almost_equal(score1, score2)
 
 
-@ignore_warnings
 def test_raises_on_score_list():
     # Test that when a list of scores is returned, we raise proper errors.
     X, y = make_blobs(random_state=0)
     f1_scorer_no_average = make_scorer(f1_score, average=None)
     clf = DecisionTreeClassifier()
-    with pytest.raises(ValueError):
+    score_kwargs = {'error_score': np.nan}
+
+    # since we're using a list of scores, our error will be the following
+    error_message = "ValueError: scoring must return a number"
+
+    # the warning message we're expecting to see
+    warning_message = ("Scoring failed. The score on this train-test "
+                       "partition for these parameters will be set to %f. "
+                       "Details: \n%s" % (score_kwargs['error_score'],
+                                          error_message))
+
+    with warnings.catch_warnings(record=True) as record:
+        warnings.simplefilter("always")
         cross_val_score(clf, X, y, scoring=f1_scorer_no_average)
+        assert len(record) > 0
+        for item in record:
+            assert 'Traceback (most recent call last):\n' in str(item.message)
+            split = str(item.message).splitlines()
+            mtb = split[0] + '\n' + split[-1]
+            assert warning_message in mtb
```

Comment:
I think the catch_warnings is useful, since I'm able to verify the ValueError as well in the traceback, not just check if a certain warning is raised. Please let me know what do you think. Please let me know if you have any comments on the patch and does it need any modifications?